feat(docs): Clarify rate limiting and trace completeness (#1301)
getsentry/develop: src/docs/dynamic-sampling/the-big-picture.mdx

</Alert>

## Rate Limiting and Trace Completeness

Dynamic sampling ensures complete traces by retaining all transactions associated with a trace if the head transaction is preserved.

> Despite dynamic sampling providing trace completeness, transactions may still be missing from a trace when rate limiting drops one or more transactions. Rate limiting drops transactions without regard for the trace, making each decision independently and potentially resulting in broken traces.

Reviewer: ```suggestion
Despite dynamic sampling providing trace completeness, transactions or other items (errors, replays, ...) may still be missing from a trace when rate limiting drops one or more transactions. Rate limiting drops items without regard for the trace, making each decision independently and potentially resulting in broken traces.
```

<Alert title="💡 Example" level="info">

> For example, if there is a trace from `Project A` to `Project B` and `Project B` is subject to rate limiting, transactions of `Project B` from the trace initiated by `Project A` are likely to be lost.

Reviewer: ```suggestion
For example, if there is a trace from `Project A` to `Project B` and `Project B` is subject to rate limiting or quota enforcement, transactions of `Project B` from the trace initiated by `Project A` are lost.
```
